embulk-core/build.gradle

```diff
@@ -23,30 +23,32 @@ repositories {
 
 // determine which dependencies have updates: $ gradle dependencyUpdates
 dependencies {
-    implementation project(":embulk-api")
-
-    implementation "com.google.guava:guava:18.0"
-    implementation "com.google.inject:guice:4.0"
-    implementation "com.google.inject.extensions:guice-multibindings:4.0"
-    implementation "javax.inject:javax.inject:1"
-    implementation "com.fasterxml.jackson.core:jackson-annotations:2.6.7"
-    implementation "com.fasterxml.jackson.core:jackson-core:2.6.7"
-    implementation "com.fasterxml.jackson.core:jackson-databind:2.6.7"
-    implementation "com.fasterxml.jackson.datatype:jackson-datatype-guava:2.6.7"
-    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.6.7"
-    implementation "com.fasterxml.jackson.datatype:jackson-datatype-joda:2.6.7"
-    implementation "com.fasterxml.jackson.module:jackson-module-guice:2.6.7"
-    implementation "org.slf4j:slf4j-api:1.7.12"
-    implementation "org.jruby:jruby-complete:" + rootProject.jrubyVersion
-    implementation "javax.validation:validation-api:1.1.0.Final"
-    implementation "org.apache.bval:bval-jsr303:0.5"
-    implementation "joda-time:joda-time:2.9.2"
-    implementation "org.msgpack:msgpack-core:0.8.11"
+    // Embulk plugins have expected that embulk-core's dependencies are declared as "compile" in pom.xml.
+    // Although those dependencies would be finally removed from embulk-core's direct dependencies,
+    // those "compile" declaration would be kept so that plugin developers won't be confused unnecessarily.
```

Comment:
This comment says this will be moved to `implementation` eventually? Do I understand correctly `removed`?

Reply:
Eventually, many of them are "removed" from this artifact `embulk-core`, and go to `embulk-deps-*` to be behind sub ClassLoaders.

Some may remain in `embulk-core`. TBD to keep `api` or back to `implementation`.